### ISSUE TITLE
The hunt for Ref October - problems on shutdown/d3d device release

### ISSUE THREAD

**Issue by WinterSnowfall** [AlpyneDreams/d8vk#97]
This is more of a meta-issue to keep track of the overall problem, so I'll break character and not follow the template for once.

There are already a number of open issues for games that crash on exit (e.g. BloodRayne, BloodRayne 2, Railroad Tycoon 3 etc.) with various errors, some outright pointing to dealocation/release problems in d8vk (see #90).

I'm also getting occasional crashes/hangs on exit with other d3d8 games. While it's hard to assess if they are caused by some other problem in Wine or d8vk, this may be the underlying cause behind them as well.

While retesting latest master, I've also noticed Duke Nukem - Manhattan Project errors out on exit, issuing the following messages from its own loggers:
```
*** ERROR *** <dx8> release refs (device)
*** ERROR *** <dx8> release refs 1 (d3d)
```
That doesn't cause any issue with the game or an outright crash, since in this particular case the game appears to handle them gracefully, however these errors do not show up with WineD3D.

P.S.: Should probably clarify that "The hunt for Ref October" is not a game, but a pun on a 90s movie & Tom Clancy novel. Since hunting for stray refs is bound to be as "thrilling", I thought it fitting.

**Comment by WinterSnowfall** [comment]
Also related: https://github.com/AlpyneDreams/d8vk/issues/75#issuecomment-1455333471

**Comment by WinterSnowfall** [comment]
I have hunted and sunk some misbehaving refs with #122 and we are now a lot more inline to what I regularly see in WineD3D traces in terms of refcounts, however there are still a few offenders, namely:

`m_renderTarget = ref(new D3D8Surface(this, std::move(pRT9)));`

`m_depthStencil = ref(new D3D8Surface(this, std::move(pStencil9)));`

Saving a ref in a COM object is bad, I've learned that much. Games like Railroad Tycoon 3 (#90) will complain about these not being released properly on device destruction. If I remove them, however, other games outright crash (#92, #57) , saying they can't get a reference to the render target or depth stencil.

I'm not sure how to fix this without breaking the other games that are dependent on these refs, however removing them does indeed fix #90, which doesn't seem to have any issues with getting its render targets and depth stencils even without the refs.

Games like Duke Nukem - Manhattan Project no longer error out on exit with #122, so at least that's some good news.
